dbt/adapters/snowflake/connections.py

```diff
@@ -260,103 +260,47 @@ def open(cls, connection):
-                        "snowflake connection. No retries "
-                        "attempted: '{}'".format(e)
-                    )
+        timeout = creds.connect_timeout
+
+        def connect():
+            handle = snowflake.connector.connect(
+                account=creds.account,
+                user=creds.user,
+                database=creds.database,
+                schema=creds.schema,
+                warehouse=creds.warehouse,
+                role=creds.role,
+                autocommit=True,
+                client_session_keep_alive=creds.client_session_keep_alive,
+                application="dbt",
+                insecure_mode=creds.insecure_mode,
+                **creds.auth_args(),
+            )
 
-                    connection.handle = None
-                    connection.state = "fail"
-
-                    raise FailedToConnectException(str(e))
-
-            except snowflake.connector.errors.Error as e:
-                if creds.retry_all and creds.connect_retries > 0:
-                    error = e
-                    logger.warning(
-                        "Got an error when attempting to open a "
-                        "snowflake connection. Retrying due to "
-                        "'retry_all' configuration set to true."
-                        "This was attempt number: {attempt} of "
-                        "{retry_limit}. "
-                        "Retrying in {timeout} "
-                        "seconds. Error: '{error}'".format(
-                            attempt=attempt,
-                            retry_limit=creds.connect_retries,
-                            timeout=creds.connect_timeout,
-                            error=e,
-                        )
-                    )
-                    sleep(creds.connect_timeout)
-                else:
-                    logger.debug(
-                        "Got an error when attempting to open a "
-                        "snowflake connection. No retries "
-                        "attempted: '{}'".format(e)
-                    )
+            if creds.query_tag:
+                handle.cursor().execute(
+                    ("alter session set query_tag = '{}'").format(creds.query_tag)
+                )
 
-                    connection.handle = None
-                    connection.state = "fail"
+            return handle
 
-                    raise FailedToConnectException(str(e))
+        def exponential_backoff(attempt: int):
+            return attempt * attempt
 
-        else:
-            logger.debug(
-                "Got an error when attempting to open a snowflake "
-                "connection: '{}'".format(error)
-            )
+        retryable_exceptions = []
+        if creds.retry_all:
+            retryable_exceptions = [snowflake.connector.errors.Error]
+        elif creds.retry_on_database_errors:
+            retryable_exceptions = [snowflake.connector.errors.DatabaseError]
```

Comment:
IMO let's use `retry_on_database_errors: True` as our default. Users can optionally turn if off, or they can opt into all errors via `retry_all`.

(@nathaniel-may may also be inspired to dig further into Snowflake docs)

Reply:
I found more sensible defaults by looking at the [connector source](https://github.com/snowflakedb/snowflake-connector-python/blob/2ccd3c811958170854ff4cac159e7c125bf8b4f3/src/snowflake/connector/errors.py).